rfc/rfc-2/2 - Deletion Info Conversion.md

```diff
@@ -0,0 +1,221 @@
+to detect new deletion vectors in the source table and add it to the conversion pipeline similar to data files. 
+Handling new deletion vectors does not require any changes in the `DataFilesDiff` class. The class encapsulates the
+addition and removal of data files, and the addition of deletion vectors is treated as a new data file addition.
+Which means, if a source produces deletion vectors, going forward, the `filesAdded` field in `DataFilesDiff` will
+not only contain data files but also deletion vectors. The consumers which cannot handle instance of 
+`InternalDeletionVector` in `filesAdded` can either ignore them or raise an exception.
+
+**Updates in `DeltaActionsConverter`**
+The converter class is responsible for converting Delta Lake `actions` into internal representations, resulting in
+`TableChange` objects. The converter is updated to extract deletion vectors from the `AddFile` actions. Currently,
+the converters read content from source metadata files and then close the file handles. However, this will be different
+for deletion vectors. To avoid the potential memory explosion, the converters will create a provider of deletion vectors
+ordinals, which will be used to stream the ordinals of the records deleted to the targets.
+
+**Changes in `DeltaConversionSource`**
+As described above, the delta commit logs adds data file entries in commit log each time delete vectors are updated. 
+This needs special handling in the delta source, to avoid duplication of data file actions.
+
+**New class `IcebergDeleteVectorConverter`**
+
+The purpose of this class is to convert internal deletion vectors to Iceberg positional delete files. The class will
+contain methods matching other Iceberg converters, for instance `toIceberg`. However, similar to unique handling of
+deletion vectors in `DeltaActionsConverter`, this class will have a unique implementation. This class will generate 
+data files for deletion vectors, which will be added to the target table. The cost of conversion of deletion vectors
+is proportional to the number of records deleted, and not the size of metadata or number of files. 
+
+Iceberg is quite
+flexible when it comes to location of data and metadata files. Since deletion vectors are data files, colocating them
+with metadata files could be confusing. Typically, deletion files are stored in partition specific directories, in
+the data directory of a table. However, it seems cleaner to keep all XTable generated data files in a separate
+directory, which will be configurable. The default location for deletion vectors will be `deletion-vectors` directory
+in the base directory of the table.
+
+**Modifications in `IcebergDataFileUpdatesSync` and `IcebergConversionTarget`**
+Finally, existing classes that handle Iceberg data files and sync operations will be updated to handle deletion vectors.
+Most changes are straightforward and inline with existing logic. However, as described earlier, new deletion vectors
```

Comment:
How will you handle the removal of deletion vectors? Will the path from the source deterministically map to some location in this new `deletion-vectors` directory?

Reply:
@ashvina let's create a placeholder issue or RFC PR and link it here with a note that this will be discussed further in that document.

Reply:
Unresolving as this is not yet addressed

Reply:
Just to clarify, are we holding off on merging the delete vector conversion until there is a concrete plan for artifact cleanup?

Reply:
No, I have simply asked that the placeholder document be linked as part of this document. It will allow readers to navigate to the follow-up more easily. It would be nice to also call out that this will cause some growth in data retained for users until that issue is resolved.

Reply:
Good point!
If I understand correctly, once the proposal is finalized and code merged, a cleaned-up version of the RFC should be uploaded to docs as a doc, including a reference to the pending cleanup task.

Reply:
Added this note: https://github.com/apache/incubator-xtable/pull/634/commits/23080cd710d53af21096389be43d228deb7e870c

Reply:
Thanks!